src/js/core/methods/CardanoSignTransaction.js

```diff
@@ -149,22 +151,28 @@ export default class CardanoSignTransaction extends AbstractMethod {
         });
     }
 
+    async filteredCall(cmd: DeviceCommands, type: MessageKey, msg: Object = {}) {
+        const res = await cmd.call(type, msg);
+        return cmd._filterCommonTypes(res);
+    }
+
     async run() {
         this._ensureFirmwareSupportsParams();
 
         const cmd = this.device.getCommands();
-        const { message } = await cmd.typedCall('CardanoSignTx', 'CardanoSignedTx', this.params);
 
-        let hash = message.tx_hash;
-        let serializedTx = message.serialized_tx;
+        let serializedTx = '';
 
-        let expectMoreChunks = !!message.expect_more_chunks;
-        while (expectMoreChunks) {
-            const { message } = await cmd.typedCall('CardanoSignedTxAck', 'CardanoSignedTx');
-            serializedTx += message.serialized_tx || "";
-            expectMoreChunks = !!message.expect_more_chunks;
+        let { message } = await this.filteredCall(cmd, 'CardanoSignTx', this.params);
+        while (message.signed_tx_chunk) {
+            serializedTx += message.signed_tx_chunk;
+            ({ message } = await this.filteredCall(cmd, 'CardanoSignedTxChunkAck'));
+        }
+        if (message.serialized_tx) {
```

Comment:
I'd add a comment here clarifying this is a backawards-compatibility adjustment made to support Trezor Firmware 2.3.6 and older where the tx was sent in a single chunk